src/manifests/manifest.py

```diff
@@ -51,15 +51,17 @@ def from_path(cls, path):
 
     @classmethod
     def compact(cls, d):
-        result = {}
-        for k, v in d.items():
-            if isinstance(v, dict):
-                nested = cls.compact(v)
-                if nested:
-                    result[k] = nested
-            elif v and v != []:
-                result[k] = v
-        return result
+        if isinstance(d, list):
+            return list(map(lambda i: cls.compact(i), d))
+        elif isinstance(d, dict):
+            result = {}
+            for k, v in d.items():
+                v = cls.compact(v)
+                if v and v != []:
```

Comment:
```suggestion
                if v:
```
should do the work here

Reply:
I wanted to capture intent. Is it true that an empty list returns false on []? Will check in a future PR. Weird.

Reply:
https://github.com/opensearch-project/opensearch-build/pull/877